Sources/AppcuesKit/Data/Models/Experience/ExperienceComponent+Sticky.swift

```diff
@@ -0,0 +1,102 @@
+            body ?? ExperienceComponent.spacer(SpacerModel(id: UUID(), spacing: nil, style: nil)),
+            stickyTopComponents.reduced(),
+            stickyBottomComponents.reduced()
+        )
+    }
+
+    /// Recursively extract sticky stacks while filtering those from the body component.
+    private func split() -> (nonSticky: ExperienceComponent?, stickyTop: [ExperienceComponent], stickyBottom: [ExperienceComponent]) {
+        var nonStickyComponents: [ExperienceComponent] = []
+        var stickyTopComponents: [ExperienceComponent] = []
+        var stickyBottomComponents: [ExperienceComponent] = []
+
+        switch self {
+        case .stack(let model) where model.sticky == .top:
+            return (nil, [self], [])
+        case .stack(let model) where model.sticky == .bottom:
+            return (nil, [], [self])
+        case .stack(let model) where model.sticky == nil:
+            model.items.forEach { item in
+                let (body, nestedTop, nestedBottom) = item.split()
+                if let body = body {
+                    nonStickyComponents.append(body)
+                }
+                stickyTopComponents.append(contentsOf: nestedTop)
+                stickyBottomComponents.append(contentsOf: nestedBottom)
+            }
+            return (
+                // Same stack, just with sticky items filtered out
+                .stack(StackModel(
+                    id: model.id,
+                    orientation: model.orientation,
+                    distribution: model.distribution,
+                    spacing: model.spacing,
+                    items: nonStickyComponents,
+                    sticky: model.sticky,
+                    style: model.style)),
+                stickyTopComponents,
+                stickyBottomComponents
+            )
+        case .box(let model):
+            model.items.forEach { item in
+                let (body, nestedTop, nestedBottom) = item.split()
+                if let body = body {
+                    nonStickyComponents.append(body)
+                }
+                stickyTopComponents.append(contentsOf: nestedTop)
+                stickyBottomComponents.append(contentsOf: nestedBottom)
+            }
+            return (
+                // Same box, just with sticky items filtered out
+                .box(BoxModel(
+                    id: model.id,
+                    items: nonStickyComponents,
+                    style: model.style)),
+                stickyTopComponents,
+                stickyBottomComponents
+            )
+        default:
+            return (self, [], [])
+        }
+    }
+}
+
+private extension Array where Element == ExperienceComponent {
+    /// Reduce multiple components into a single vertical stack.
+    func reduced() -> ExperienceComponent? {
+        switch count {
+        case 0:
+            return nil
+        case 1:
+            return self[0]
```

Comment:
On Android, I always wrapped the sticky top/bottom regions in a vstack, if not empty, even in the single item case. This would be a bit more efficient in terms of overall depth of layout tree. I think we can update Android to match.